vendor/bat-native-ledger/src/bat/ledger/internal/legacy/bat_helper.cc

```diff
@@ -12,83 +12,78 @@
-  rapidjson::Document d;
-  d.Parse(json.c_str());
-
-  // has parser errors or wrong types
-  bool error = d.HasParseError() || (false == d.HasMember(fieldName.c_str()));
-  if (!error) {
-    *value = d[fieldName.c_str()].GetString();
+  absl::optional<base::Value> result =
+      base::JSONReader::Read(json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                                       base::JSONParserOptions::JSON_PARSE_RFC);
+  if (!result || !result->is_dict())
+    return false;
+
+  if (auto* field = result->GetDict().FindString(field_name)) {
+    *value = *field;
+    return true;
   }
-  return !error;
+
+  return false;
 }
 
 bool getJSONTwitchProperties(
     const std::string& json,
     std::vector<base::flat_map<std::string, std::string>>* parts) {
-  rapidjson::Document d;
-  d.Parse(json.c_str());
-
-  // has parser errors or wrong types
-  bool error = d.HasParseError();
-  if (!error) {
-    for (auto & i : d.GetArray()) {
-      const char * event_field = "event";
-      base::flat_map<std::string, std::string> eventmap;
-
-      auto obj = i.GetObject();
-      if (obj.HasMember(event_field)) {
-        eventmap[event_field] = obj[event_field].GetString();
-      }
-
-      const char * props_field = "properties";
-      if (obj.HasMember(props_field)) {
-        eventmap[props_field] = "";
-
-        const char * channel_field = "channel";
-        if (obj[props_field].HasMember(channel_field) &&
-          obj[props_field][channel_field].IsString()) {
-          eventmap[channel_field] = obj[props_field][channel_field].GetString();
-        }
-
-        const char * vod_field = "vod";
-        if (obj[props_field].HasMember(vod_field)) {
-          eventmap[vod_field] = obj[props_field][vod_field].GetString();
-        }
-
-        const char * time_field = "time";
-        if (obj[props_field].HasMember(time_field)) {
-          double d = obj[props_field][time_field].GetDouble();
-          eventmap[time_field] = std::to_string(d);
-        }
-      }
-      parts->push_back(eventmap);
+  absl::optional<base::Value> result =
+      base::JSONReader::Read(json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                                       base::JSONParserOptions::JSON_PARSE_RFC);
+  if (!result || !result->is_list())
+    return false;
+
+  for (auto& item : result->GetList()) {
+    auto& dict = item.GetDict();
+    base::flat_map<std::string, std::string> eventmap;
+    if (auto* value = dict.FindString(kFieldEvent)) {
+      eventmap[kFieldEvent] = *value;
+    }
+
+    if (auto* properties = dict.FindDict(kFieldProperties)) {
+      eventmap[kFieldProperties] = "";
+
+      if (auto* channel = properties->FindString(kFieldChannel))
+        eventmap[kFieldChannel] = *channel;
+
+      if (auto* vod = properties->FindString(kFieldVod))
+        eventmap[kFieldVod] = *vod;
+
+      if (auto time = properties->FindDouble(kFieldTime))
+        eventmap[kFieldTime] = std::to_string(*time);
```

Comment:
We should use `base::NumberToString()`.

Reply:
Fixed in e2c74682957f896d5481ee3e371a29b17b7b7929.